### ISSUE TITLE
GUI independent API?

### ISSUE THREAD

**Issue by avanov** [Gabriel439/Haskell-Typed-Spreadsheet-Library#22]
Hi! I wonder if this work can be detached from GTK UI and provided as an API on pure data structures such as heterogeneous collections/open sums? I think what I'm looking for is something like an improved and enhanced version of https://github.com/Gabriel439/Haskell-MVC-Updates-Library :)

**Comment by Gabriella439** [comment]
@avanov: Yeah, the underlying `Cell` type could be separated into a different package.  I don't remember the exact reason why I didn't do this at the time, but I *think* the reason why is because I wasn't sure exactly what the requirements for the standalone package would be, so I wasn't sure what balance to strike between the simpler `Cell` type from `typed-spreadsheet` and the more complicated `Updatable` type from `mvc-updates`.  Keeping the `Cell` type linked to the `typed-spreadsheet` package made it easier to narrow down its purpose.

**Comment by avanov** [comment]
Makes sense, thanks! I'll play with it whilst trying to apply the model to my use cases and maybe it'll naturally manifest an API for it. Let's keep this issue open, if you don't mind. There may be other users with similar ideas who could suggest their vision of it.

Are you planning to support `mvc-updates` in a long run, or do you see it potentially substituted by a subset of `typed-spreadsheet`?

**Comment by Gabriella439** [comment]
`mvc-updates` is in "maintenance mode" meaning that I'm not actively developing it since there doesn't seem to be enough interest around it, but I'll still respond to issues with some limited work and still review/merge pull requests
